arch/posix/include/posix_soc_if.h

```diff
@@ -35,6 +33,16 @@ void posix_irq_full_unlock(void);
 int  posix_get_current_irq(void);
 /* irq_offload() from irq_offload.h must also be defined by the SOC or board */
 
+static inline unsigned int z_arch_irq_lock(void)
```

Comment:
I would not define these here, but maybe in include/arch/posix/arch.h 
(both for consistency with other archs, and because this header is meant to define the interface a posix soc provides to the posix arch)

If you are doing this change, I would then also be consistent and do the same for the other z_arch_irq* functions here. I think it makes sense if I send a quick PR and then if you want you can take that commit into this one.

Reply:
@andrewboie I just added an extra patch on top of this branch in here:
https://github.com/aescolar/zephyr/commits/pa_hide_internal_if 
https://github.com/aescolar/zephyr/commit/92671a6a03395ca55fb084da90c33cad596c36f9
Feel free to take it in if you agree. Or tell how you would prefer to handle it otherwise.

Reply:
I pulled your patch into the series.

Reply:
Thanks @andrewboie . This new commit is indeed going a bit further moving things around, but as this arch API changes are also causing extra merge work for off-tree POSIX boards I thought it better to take them in one go, and put those inlines in a more proper place. It should also simplify a bit future arch interface cleanup PRs (so you dont need to dig thru the posix SOC and boards files)